src/coreclr/interpreter/compiler.cpp

```diff
@@ -1411,6 +1611,126 @@ bool InterpCompiler::CreateBasicBlocks(CORINFO_METHOD_INFO* methodInfo)
+            {
+                if ((clause.Flags == CORINFO_EH_CLAUSE_NONE) || (clause.Flags == CORINFO_EH_CLAUSE_FILTER))
+                {
+                    pBB->clauseType = BBClauseCatch;
+                }
+                else
+                {
+                    assert((clause.Flags == CORINFO_EH_CLAUSE_FINALLY) || (clause.Flags == CORINFO_EH_CLAUSE_FAULT));
+                    pBB->clauseType = BBClauseFinally;
+                }
+            }
+        }
+
+        if (clause.Flags == CORINFO_EH_CLAUSE_FILTER)
+        {
+            if ((codeStart + clause.FilterOffset) > codeEnd)
+                return false;
+
+            // The filter funclet is always stored right before its handler funclet.
+            // So the filter end offset is equal to the start offset of the handler funclet.
+            InterpBasicBlock* pFilterBB = GetBB(clause.FilterOffset);
+            pFilterBB->isFilterOrCatchFuncletEntry = true;
+            pFilterBB->clauseVarIndex = m_clauseVarsIndex + i;
+
+            // Initialize the filter stack state. It initially contains the exception object.
+            pFilterBB->stackHeight = 1;
+            pFilterBB->pStackState = (StackInfo*)AllocMemPool(sizeof (StackInfo));
+            pFilterBB->pStackState[0].type = StackTypeO;
+            pFilterBB->pStackState[0].size = INTERP_STACK_SLOT_SIZE;
+            pFilterBB->pStackState[0].clsHnd = NULL;
+            pFilterBB->pStackState[0].var = pFilterBB->clauseVarIndex;
+
+            // Find and mark all basic blocks that are part of the filter region.
+            for (uint32_t j = clause.FilterOffset; j < clause.HandlerOffset; j++)
+            {
+                InterpBasicBlock* pBB = m_ppOffsetToBB[j];
+                if (pBB != NULL && pBB->clauseType == BBClauseNone)
+                {
+                    pBB->clauseType = BBClauseFilter;
+                }
+            }
+        }
+        else if (clause.Flags == CORINFO_EH_CLAUSE_FINALLY|| clause.Flags == CORINFO_EH_CLAUSE_FAULT)
+        {
+            InterpBasicBlock* pFinallyBB = GetBB(clause.HandlerOffset);
+
+            // Initialize finally handler stack state to empty.
+            pFinallyBB->stackHeight = 0;
+        }
+        
+        if (clause.Flags == CORINFO_EH_CLAUSE_NONE || clause.Flags == CORINFO_EH_CLAUSE_FILTER)
+        {
+            InterpBasicBlock* pCatchBB = GetBB(clause.HandlerOffset);
+            pCatchBB->isFilterOrCatchFuncletEntry = true;
+            pCatchBB->clauseVarIndex = m_clauseVarsIndex + i;
+
+            // Initialize the catch / filtered handler stack state. It initially contains the exception object.
+            pCatchBB->stackHeight = 1;
+            pCatchBB->pStackState = (StackInfo*)AllocMemPool(sizeof (StackInfo));
+            pCatchBB->pStackState[0].type = StackTypeO;
+            pCatchBB->pStackState[0].size = INTERP_STACK_SLOT_SIZE;
+            pCatchBB->pStackState[0].var = pCatchBB->clauseVarIndex;
```

Comment:
It looks like for filtered catches we reuse the same var for the exc var in both the filter and the catch. Is this going to cause any problems for liveness or other analyses we do in the interpreter, like offset allocation?